### PR TITLE
mobile: ensuring the non-listener build works correctly

### PR DESCRIPTION
**Pull request by alyssawilk**
Changing one of our CI builds to remove a listener file, to ensure the downstream listener code doesn't leak back into the envoy_mobile_listener=disabled build.

Also removing a spurious library include so the build passes.

Risk Level: low
Testing: CI update
Docs Changes: n/a
Release Notes: n/a
Part of https://github.com/envoyproxy/envoy/pull/25852

### CODE REVIEW

File: .github/workflows/core.yml [hunk 1]
```diff
@@ -20,6 +20,8 @@ jobs:
       image: envoyproxy/envoy-build-ubuntu:458cb49ca2013c0ccf057d00ad1d4407920c4e52
     steps:
     - uses: actions/checkout@v3
+    - name: Ensure no listener leaks
+      run: rm source/extensions/listener_managers/listener_manager/listener_manager_impl.h
```
Comment (RyanTheOptimist) [inline .github/workflows/core.yml:24]: HAH!

**Review by RyanTheOptimist — APPROVED** [review]
Very clever. A bit of a big hammer, but it makes sense!

**Comment by jpsim** [comment]
+1 thanks for the fix, although the way we're validating it is a bit hacky.

Bazel has ways to test/validate the final binary as part of building. One thing we could do is assert that certain symbols aren't in the binary when building with listeners disabled. We do that at Lyft when building our iOS apps. I'll spend a few minutes to see if I can hook that up here.

**Comment by jpsim** [comment]
> I'll spend a few minutes to see if I can hook that up here.

I spent a few minutes and got stuck in a few places. For one, we use `apple_verification_test` for verifying the presence/absence of symbols so we can't use that on Linux. I also ran into a few issues even on macOS hosts, so I'm comfortable sticking with the current solution of nuking a source file that shouldn't be compiled.

**Comment by alyssawilk** [comment]
yeah I knew I could do it "more cleanly" with a bazel query but I trust rm -rf more :-)

**Pull request merged**